### DISCUSSION TITLE
Lesson 13: Disconnect Metamask does not work

### DISCUSSION THREAD

**Discussion by nhache** [smartcontractkit/full-blockchain-solidity-course-py#217] [Q&A] [answered]
Hi all,

in lesson 13 (YT 14:27) clicking disconnect won't actually disconnect Metamask (at least not for me in Brave browser). 

I tried both the version in the video:
`{isConnected ? (
                <Button color="primary" onClick={deactivate}>
                    Disconnect
                </Button>
            ) : (
                <Button
                    color="primary"
                    onClick={() => activateBrowserWallet()}
                >
                    Connect
                </Button>
            )}`
as well as the version from the [docs](https://usedapp.readthedocs.io/en/latest/guide.html):
`{account && <button color = "primary" onClick={deactivate}>
                Disconnect
            </button>}
            {!account && <button color="primary" onClick={() => activateBrowserWallet()}>
                    Connect 
            </button>}`

Funnily enough, I also went and checked out the [example that useDapp is leveraging](https://example.usedapp.io/balance) and the same problem appears there. That's definitely not expected?!

**Comment by KennethJChew** [comment]
I have this issue too, did you manage to get the disconnect button working?

**Reply by nhache** [reply]
Hey Kenneth, sorry for the late reply. Didn't, doesn't seem to be supported by useDapp. Actually, I just went ahead with vanillaJS, it's simple enough with `window.ethereum`. Super straightforward, the [docs](https://docs.metamask.io/guide/ethereum-provider.html) are very good as well with a bunch of boilerplate so I'd recommend that.

**Comment by PatrickAlphaC** [comment, marked as answer]
You're right. This was an issue in the implementation. Sorry!

**Reply by nhache** [reply]
Thanks for confirming Patrick!
